duqtools/operations.py

```diff
@@ -0,0 +1,145 @@
+from __future__ import annotations
+
+import logging
+from collections import deque
+from inspect import signature
+from typing import Callable
+
+import click
+from pydantic import Field
+
+from .schema import BaseModel
+
+logger = logging.getLogger(__name__)
+
+
+class Operation(BaseModel):
+    """Operation, simple class which has a callable action."""
+
+    description: str = Field(
+        description='description of the operation to be done')
+    action: Callable = Field(
+        description='a function which can be executed when we '
+        'decide to apply this operation')
+    args: tuple = Field((),
```

Comment:
Better to use `None` here, and check against that.
```suggestion
    args: tuple = Field(None,
```

Reply:
I have tried that , however I'd need to use `Optional` then (not a problem), but more importantly, the `__call__` would have to check for existence of `args`. 
Passing an empty tuple directly is fine, so  I like to keep it as is, but maybe I am missing something else?

Reply:
I was thinking of this:
https://docs.python-guide.org/writing/gotchas/#what-you-wrote

It only applies to mutable objects, which tuple isn't, but the dict below is. So in general I think it is clearer to set `None` as the default and handle it in the code instead.